lib/apiservers/service/restapi/handlers/vch_list_get.go

```diff
@@ -98,29 +98,39 @@ func listVCHs(op trace.Operation, d *data.Data, validator *validate.Validator) (
 func vchsToModels(op trace.Operation, vchs []*vm.VirtualMachine, executor *management.Dispatcher) []*models.VCHListItem {
 	installerVer := version.GetBuild()
 	payload := make([]*models.VCHListItem, 0)
+
 	for _, vch := range vchs {
 		var version *version.Build
 		var dockerHost string
 		var adminPortal string
 		name := path.Base(vch.InventoryPath)
 		id := vch.Reference().Value
 
-		if vchConfig, err := executor.GetNoSecretVCHConfig(vch); err == nil {
+		vchConfig, err := executor.GetNoSecretVCHConfig(vch)
+		// If we can't get the extra config from this VCH, the VCH at this point could already been deleted / partially created / corrupted
+		// we ignore this partial VCH, log the error and skip to next one
+		if err != nil {
+			op.Warnf("Failed to get extra config from VCH %s: %s", vch.Reference(), err)
+		} else {
```

Comment:
I would not use else here for readability purposes. I would use "continue" in case of an error, so less indentations.